fhir-server-test/src/test/java/org/linuxforhealth/fhir/server/test/ServerSpecTest.java

```diff
@@ -696,6 +712,82 @@ public void testConditionalUpdateObservation2() throws Exception {
                 "Input resource 'id' attribute must match the id of the search result resource");
 
     }
+    
+    /**
+     * Test conditional update when :
+     * 1. No matches, no id provided.
+     * 2. No matches, id provided and doesn't already exist and update/create is enabled.
+     * 3. No matches, id provided and doesn't already exist and update/create is not enabled. 
+     * 4. No matches, id provided and id already exists.
+     * @throws Exception
+     */
+    @Test(groups = { "server-spec" }, dependsOnMethods = { "testConditionalCreateObservation" })
+    public void testConditionalUpdateObservation3() throws Exception {
+        String fakePatientRef = "Patient/" + UUID.randomUUID().toString();
+        String obsId = UUID.randomUUID().toString();
+        Observation obs = TestUtil.readLocalResource("Observation1.json");
+        Observation obsWithNoId = obs.toBuilder()
+                .subject(Reference.builder().reference(string(fakePatientRef)).build())
+                .build();
+        Observation obsWithId = obs.toBuilder()
+                .subject(Reference.builder().reference(string(fakePatientRef)).build())
+                .id(obsId)
+                .build();
+        Observation obsWithExistingId = obs.toBuilder()
+                .subject(Reference.builder().reference(string(fakePatientRef)).build())
+                .id(savedObservation.getId())
+                .build();
```

Comment:
there's no need to use the static "string" factory method any longer
```suggestion
        Observation obsWithNoId = obs.toBuilder()
                .subject(Reference.builder().reference(fakePatientRef).build())
                .build();
        Observation obsWithId = obs.toBuilder()
                .subject(Reference.builder().reference(fakePatientRef).build())
                .id(obsId)
                .build();
        Observation obsWithExistingId = obs.toBuilder()
                .subject(Reference.builder().reference(fakePatientRef).build())
                .id(savedObservation.getId())
                .build();
```

Reply:
This has been fixed as suggested.